cmake/modules/FindTar.cmake

```diff
@@ -0,0 +1,39 @@
+# Filename: FindTar.cmake
```

Comment:
Given the stupidly unfriendly name to be installing to a user's global command namespace, yes please, let's remove patching support.

If someone's interested in patching we should instead guide them toward either bsdiff or VCDIFF. Panda supporting binary patching seems like more Imagineering NIH syndrome cruft.

Reply:
![image](https://user-images.githubusercontent.com/194842/64416529-9a5dfa00-d097-11e9-9507-3d86321e7420.png)
aef81aceab51dee73b94dd48a1a30190b6d432d4 8b8796e9df6a302b7ec076b51e326f7f78c17560
Please do the needful and remove the tar stuff from CMake.  I haven't yet removed the Patchfile class since it's a public API, but we should definitely deprecate it.